src/Microsoft.ML.AutoML/CodeGen/type_converter_search_space.json

```diff
@@ -9,6 +9,10 @@
     {
       "name": "InputColumnNames",
       "type": "strings"
+    },
+    {
+      "name": "TargetType",
+      "type": "DataKind"
```

Comment:
If you want to define a custom `DataKind`, you'll need to modify automl codegen as well. Otherwise, the source generator will throw an error.

You can refer to how `bertArchitecture' on how to do it.